### PR TITLE
add MFT-SEMA100 board

### PR DESCRIPTION
**Pull request by EternAlmox**

### CODE REVIEW

**Comment by EternAlmox** [comment]
Test Setup
Board

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef-bl.dat [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+PA12 OTG_FS_DP OTG1
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+
+PE3  BMI088_A_CS CS
+PE4  BMI088_G_CS CS
+
+# Add CS pins to ensure they are high in bootloader
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef-bl.dat:48]: these are all commented out, better to remove them

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+
+USB_STRING_MANUFACTURER "MFT"
+USB_STRING_PRODUCT "MFT-SEMA100"
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART3 UART5 UART7 UART8
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:29]: adding OTG2 at the end to give dual USB endpoints is normally worthwhile, helpful for CAN diagnostics

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+# I2C4 is External
+PD12 I2C4_SCL I2C4
+PD13 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C2 I2C3 I2C4 I2C1
+
+
+define HAL_I2C_INTERNAL_MASK 1
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:88]: this should be a mask of what buses are internal only. looks like only I2C3 is internal-only? If that is correct then a mask of 2 would be right

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+
+
+
+# UARTs
+
+
+
+#USART1 TX RX 1
+PA10 USART1_RX USART1 NODMA
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:107]: why are you disabling DMA?

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+PA9 USART1_TX USART1 NODMA
+
+
+# USART2 is TX RX 2
+PD6 USART2_RX USART2 NODMA
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:112]: why disable DMA?

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+
+# IMU
+IMU BMI088 SPI:bmi088_a SPI:bmi088_g ROTATION_PITCH_180_YAW_270
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 7
+
+# compasses
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+COMPASS LIS3MDL I2C:ALL_EXTERNAL:0x1C true ROTATION_NONE
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:212]: is this a builtin compass? if its attached externally it shouldn't be in the hwdef.dat, it should be auto-detected

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+PC13 LED_BOOTLOADER OUTPUT HIGH GPIO(0)
+PC6 LED_ACTIVITY OUTPUT LOW GPIO(1) # optional
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+
+
+
+# enable RAMTROM parameter storage
+STORAGE_FLASH_PAGE 14
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:228]: is it ramtron for storage or flash? comment doesn't match the code

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+DMA_PRIORITY ADC* SPI4* TIM*UP* 
+DMA_NOSHARE  SPI4* TIM*UP*
+
+define DEFAULT_NTF_LED_TYPES 743
+
+define DEFAULT_SERIAL4_BAUD 38
+define HAL_BARO_ALLOW_INIT_NO_BARO
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:240]: no baro available??

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat (hunk 8 above)
Comment (EternAlmox) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:240]: It was from a test =) there was no baro at the begining but now it has. removing it

**Review by tridge — DISMISSED** [review]
getting closer, thanks.
please add a README.md with a description of the board in the hwdef directory, and an image.  @Hwurzburg uses this to help write documentation

**Comment by Hwurzburg** [comment]
look at this PR for an example of readme.md, images, info required for documentation: (xls doc not required) https://github.com/ArduPilot/ardupilot/pull/26854
also how commits are organized
it is usually a good idea to make the board_types.txt a separate PR to reserve the board id since it might be taken while this PR gets finalized and documentation provided....avoids having thee change your board id at the last minute when this gets about to be merged

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+# drdy pins
+PE9  DRDY1_LIS3MDL INPUT
+
+
+# UARTs
+#USART1 TX RX 1
+PA10 USART1_RX USART1 
+PA9 USART1_TX USART1 
+
+# USART2 is TX RX 2
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:103]: This UART is not declared in Serial Order

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+define HAL_STORAGE_SIZE 32768
+
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+DMA_PRIORITY ADC* SPI4* TIM*UP* 
+DMA_NOSHARE  SPI1* SPI2* SPI4* TIM*UP*
+
+define DEFAULT_NTF_LED_TYPES 743
+define DEFAULT_SERIAL4_BAUD 38
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:223]: dcuplicate in params....and SERIAL4 is defaulted to GPS protocol and baudrate is ignored anyway...if you have specific protocol target for Serial 3 and 4 other than GPS, their current default then perhaps changing default baud rate might make sense....but not currently

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# setup the temperature compensation
+BRD_HEAT_TARG 45
+BRD_HEAT_P 50
+BRD_HEAT_I 0.07
+
+SERIAL3_BAUD 38
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm:6]: see comment on last line of hwdef....not sure why these need to be set...as is, its already the default for GPS (which is autobauded anyway)

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm (hunk 11 above)
Comment (EternAlmox) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm:6]: Weird thing that when I check for the serials from MP the default Serial value was 230 for both Serıal 3 and 4 thats added to fix it. I can remove it if it is not necessary to show the default Baud as 38

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat (hunk 10 above)
Comment (EternAlmox) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:223]: Weird thing that when I check for the serials from MP the default Serial value was 230 for both Serıal 3 and 4 thats added to fix it. I can remove it if it is not necessary to show the default Baud as 38

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm (hunk 11 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm:6]: 230 is the new default for GPS protocols....it was changed from 38 just reflect what the GPS driver actually uses....38 was not used...

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat (hunk 10 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat:223]: see below

File: libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm (hunk 11 above)
Comment (EternAlmox) [reply libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/defaults.parm:6]: Alright now that makes sense =) Fixing it.

**Comment by Hwurzburg** [comment]
@EternAlmox will review docs tomorrow...having hospital procedure done today

**Comment by Hwurzburg** [comment]
Started my review, but before I can complete it I need the mapping from the marked TXn/RXn pins on the board to the CPU's uarts...the docs are inconsistent....please provide a list and then I can finish the review

also, the current definition leaves no UART with full DMA on TX and RX so some RC protocols will not be supported (CRSF,ELRS,etc.)...and the second GPS does not have DMA, requiring ALL GPSes to be restricted to 115K, this needs to be changed

**Pull request closed without merging**